some-service/client.go

```diff
@@ -0,0 +1,70 @@
+// Package some_service
+package some_service
+
+import (
+	"GithubSearch/types/v1/api"
+	"GithubSearch/utils"
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"github.com/flannel-dev-lab/cyclops/v2/logger"
+	"net/http"
+	"time"
+)
+
+const hostname = "http://localhost"
+
+type SomeService interface {
+	ResultParser(ctx context.Context, searchResponse *api.SearchResponse) error
+}
+
+type Client struct {
+	httpClient *http.Client
+}
+
+func New() SomeService {
+	return &Client{
+		httpClient: &http.Client{
+			Timeout: 10 * time.Second,
+		},
+	}
+}
```

Comment:
Return you client, not the pointer and take the addr here

```suggestion
func New(addr string) *Client {
	return &Client{
		addr: addr,
		httpClient: &http.Client{
			Timeout: 10 * time.Second,
		},
	}
}
```